### PR TITLE
Fix #6458: DatePicker defaults before super.encodeEnd

### LINKED ISSUE #6458: DatePicker: showTime not detected automatically for LocalDateTime
In the migration document (8->9), it is stated that the attribute showTime is detected automatically based on value-binding to LocalDateTime. I cannot nor verify this for the latest PF9-snapshot. Could somebody please check and fix this. Many thanks in advance.

### PR DESCRIPTION
**Pull request by melloware**
Basically what was happening was in `BaseCalendarRenderer.encodeEnd` decisions were being made with configs before calling `DatePicker.encodeMarkup`.

So we had to move this default setting to the `encodeEnd` before calling `super.encodeEnd` so the proper values for `showTime`, `showSeconds` etc are calculated before it tries to build the default markup date.

@christophs78 @jepsar 

I have tested both the Showcase and the provided test case from user.  We also have our Primefaces Integration Tests that will run.

I will add an Integration Test case there.

### CODE REVIEW

File: src/main/java/org/primefaces/component/datepicker/DatePickerRenderer.java [hunk 1]
```diff
@@ -54,12 +54,38 @@
 
     @Override
     public void encodeEnd(FacesContext context, UIComponent component) throws IOException {
-        DatePicker datePicker = (DatePicker) component;
+        DatePicker datepicker = (DatePicker) component;
```
Comment (tandraschko) [inline src/main/java/org/primefaces/component/datepicker/DatePickerRenderer.java]: why did you rename? its actually datePicker

Comment (melloware) [reply src/main/java/org/primefaces/component/datepicker/DatePickerRenderer.java]: Its mixed throughout the file with datepicker and datePicker i will fix it every where to be datePicker.

**Comment by melloware** [comment]
Integration Test updated: https://github.com/primefaces-extensions/primefaces-integration-tests/commit/d014a7ec8d0553e95f4b14fb5b856f50b616e1af

**Comment by melloware** [comment]
OK i renamed lowercase `datepicker` to proper `datePicker` throughout the whole file.

**Comment by christophs78** [comment]
Makes sense. Thank´s for analyzing!

**Comment by melloware** [comment]
New Integrations tests as now passing after this fix!

**Comment by jepsar** [comment]
Merged to https://github.com/primefaces/primefaces/pull/6450
